mgnify_pipelines_toolkit/analysis/shared/dwc_summary_generator.py

```diff
@@ -0,0 +1,235 @@
+import argparse
+from collections import defaultdict
+import pathlib
+import logging
+import requests
+
+import pandas as pd
+import pyfastx
+
+logging.basicConfig(level=logging.DEBUG)
+
+url = "https://www.ebi.ac.uk/ena/portal/api/search?result"
+runs_url = f"{url}=read_run&fields=secondary_study_accession%2Csample_accession&limit=10&format=json&download=false"
+samples_url = f"{url}=sample&fields=lat%2Clon%2Ccollection_date%2Cdepth&limit=10&format=json&download=false"
+headers = {"Accept": "application/json"}
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--input_path", required=True, type=str, help="Input")
+    parser.add_argument(
+        "-r",
+        "--runs",
+        required=True,
+        type=str,
+        help="CSV file containing successful analyses generated by the pipeline",
+    )
+    parser.add_argument("-o", "--output", required=True, type=str, help="Output")
+
+    args = parser.parse_args()
+
+    input_path = args.input_path
+    runs = args.runs
+    output = args.output
+
+    return input_path, runs, output
+
+
+def get_metadata_from_run_acc(run_acc):
+
+    query = f"{runs_url}&includeAccessions={run_acc}"
+    res_run = requests.get(query, headers=headers)
+
+    if res_run.status_code != 200:
+        logging.warning(f"Data not found for run {run_acc}")
+        return False
+
+    sample_acc = res_run.json()[0]["sample_accession"]
+
+    query = f"{samples_url}&includeAccessions={sample_acc}"
+    res_sample = requests.get(query, headers=headers)
+
+    full_res_dict = res_run.json()[0] | res_sample.json()[0]
+
+    fields_to_clean = ["lat", "lon", "depth"]
+
+    for field in fields_to_clean:
+        val = full_res_dict[field]
+        if val == "":
+            full_res_dict[field] = "NA"
+
+    if full_res_dict["collection_date"] == "":
+        full_res_dict["Year"] = "NA"
+        full_res_dict["Month"] = "NA"
+        full_res_dict["Day"] = "NA"
+    else:
+        split_date = full_res_dict["collection_date"].split("-")
+
+        if len(split_date) == 3:
+            full_res_dict["Year"] = split_date[0]
+            full_res_dict["Month"] = split_date[1]
+            full_res_dict["Day"] = split_date[2]
+        elif len(split_date) == 2:
+            full_res_dict["Year"] = split_date[0]
+            full_res_dict["Month"] = split_date[1]
+            full_res_dict["Day"] = "NA"
+        elif len(split_date) == 1:
+            full_res_dict["Year"] = split_date[0]
+            full_res_dict["Month"] = "NA"
+            full_res_dict["Day"] = "NA"
+
+    del full_res_dict["collection_date"]
+
+    res_df = pd.DataFrame(full_res_dict, index=[0])
+    res_df.columns = [
+        "RunID",
+        "SampleID",
+        "StudyID",
+        "decimalLongitude",
+        "depth",
+        "decimalLatitude",
+        "Year",
+        "Month",
+        "Day",
```

Comment:
I wonder if a pythonic lookup exists already, for the full DwC terms list from https://github.com/tdwg/dwc ? It would be nice to do `dwc.terms.RUN_ID` or whatever.